[ADD] Project: New page about visibility and collaboration

guga-odoo opened the pull request:
task-4946930

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/documentation/pull/14040.png)](https://mergebot.odoo.com/odoo/documentation/pull/14040)

auva-odoo commented on content/applications/services/project/project_management/visibility_collaboration.rst:
```diff
@@ -0,0 +1,56 @@
+icon next to their name, and click :guilabel:`Save`: to confirm.
+
+.. note::
+   In any case, an internal user with no project access rights can still access a task, provided
+   that they are given the corresponding URL (and that they are part of the followers if the project
+   is private).
```
```suggestion
   Internal users without access to the project can still access a task if the URL has been shared with them. For private projects, they must also be a follower of the task.
```
- I'm not sure what you meant by "project access rights" but I would not use the word "access right" unless it is an actual access right (as in: https://www.odoo.com/documentation/master/applications/general/users/access_rights.html) ; feel free to change this suggestion if it is not what you meant.
- I would also clarify what you mean "by private project", I'm not really sure?

auva-odoo reviewed (changes requested):
Thank you for your work @guga-odoo ! Since this feature is relatively minor and, as I understand it, part of the standard project settings, I’d suggest adding it to the Project Management page (as a sub-section of *Configuraiton*) rather than creating a new one. My review comments reflect that approach (e.g. we won't need to repeat how to access the project settings) but let's discuss this if you see it differently!

guga-odoo commented on content/applications/services/project/project_management/visibility_collaboration.rst:
```diff
@@ -0,0 +1,56 @@
+icon on a project’s card, and click on :guilabel:`Settings` in the drop-down menu. Then click on the
+:guilabel:`Settings` tab and choose among the three :guilabel:`Visibility` options:
+
+- :guilabel:`Invited internal users (private)`: When following a project, internal users will have
+  access to all of its tasks without distinction. Otherwise, they will only have access to the
+  specific tasks they are following. A user with the Project app administrator access rights level
+  can still access the project and its tasks, even if they are not listed as followers.
```
You got it right! 
Also, sorry about the future tense. I thought the tool tips in the apps were well phrased and saw an opportunity to streamline the doc with the app tool tips, but then I forgot about the tense. Thanks for the reminder!

guga-odoo commented on content/applications/services/project/project_management/visibility_collaboration.rst:
```diff
@@ -0,0 +1,56 @@
+Inviting external users
+=======================
+
+When selecting :guilabel:`Invited portal users and all internal users (public)`, a
+:guilabel:`Share Project` button appears on top of the project’s settings page. Click on it to open
+a pop-up window that offers two options to give external users access to the project:
```
Agreed, thanks for the reminders!

guga-odoo commented on content/applications/services/project/project_management/visibility_collaboration.rst:
```diff
@@ -0,0 +1,56 @@
+   - :guilabel:`Edit`: Collaborators can view and edit all tasks in the Kanban view and choose which
+     tasks they want to follow.
+
+To revoke an invited collaborator's access, click the :icon:`fa-trash-o` :guilabel:`(fa-trash-o)`
```
Good catch, I added the Share Project part :)

guga-odoo commented:
Hi @auva-odoo ! I have deleted the page I originally created and added the Collaboration and Visibility section on the Project Management page as you suggested. However I haven't added it as a sub category withing the Settings sub-section, but on the same (heading) level as it makes more sense to me considering how we organized the other sub-sections (scheduling activities and top bar). What do you think?

auva-odoo reviewed (changes requested):
Nice work @guga-odoo, thank you for the changes @guga-odoo :slightly_smiling_face: ! I only have two super minor comments (and one of them is more of an FYI), besides this one:

Regarding the section's heading, I don’t think it should be an H2; to me, *Visibility and collaboration* feels like part of the project’s standard setup (as opposed to *optional* configuration that is dependent on the installation of another app, for example) so it makes more sense under the Configuration heading. Things like Scheduling activities and Top bar seem more about actually working with the project, not configuring it per say. We can discuss this more if you'd like :wink:
